### DISCUSSION TITLE
Support For windows and updated bootloader

### DISCUSSION THREAD

**Discussion by Biswarup-exe** [PinguinoIDE/pinguinoide.github.io#32] [Pinguino IDE]
Hi, I am trying to build a PINGUINO board using pic18f4550 microcontroller,
Can anyone please share an updated bootloader .HEX file and IDE installation for windows if available.

**Comment by swhomans** [comment]
Hi, I just noticed that you have not received a reply to your enquiry. I'm not one of the developers but I have some experience with Pinguino. I have built several 18f4550 boards. As far as I am aware the developers have not yet released an updated version of Pinguino for Windows. As I'm sure you will have noted, install files are available for Linux. With some readily solvable issues, I have been able to run the latest version under Ubuntu 20.04, and both 8 bit and 32 bit code compiles and uploads without problems.  I am a windows user too, so I run Ubuntu from windows using VirtualBox. It is very easy to get Linux running in this way, and I suggest your best option if you want to use the latest code is to switch to Linux. However, if you want to use windows, I can send you a .zip file containing Pinguino X.4 rev 959. This is obviously not the latest code, but I have used it for a number of years to run some rather complex code on both P8 and P32. I recently upgraded to Windows 11 and this version installs with ease.

**Reply by Biswarup-exe** [reply]
Hello, Thank you for Your response.
It would be great if you can share the zip file for Pinguino X.4 revp59.

**Comment by swhomans** [comment]
OK no problem. In fact the relevant file is still available online via Google code: https://code.google.com/archive/p/pinguino32/downloads.
You need the file "Pinguino-IDE-snapshot-i386-unknown-win32-20131209-rev959.7z". Download the file and then proceed as follows:

    1. Using 7-Zip, extract the file to an arbitrary folder. It's important to extract to a folder whose name does not incl;ude   spaces.
    2. Move the x4-easy-rev959 folder to e.g. C:\x4 (Because gpasm.exe does not recognise a long folder name.) 
    3. Execute C:\x4-easy-revNNN\pinguino.exe,or C:\x4-easy-revNNN\pinguino_start.bat
    
That should be all that's required. Let me know if you have any problems. Good luck!

**Reply by Biswarup-exe** [reply]
Hello, I was able to install it, now for bootloader  I will be uploading these .hex files ->https://github.com/PinguinoIDE/pinguino-bootloaders/tree/master/p8/usb/v4.x/hex. Hope these will work.

And this is my installation screenshot
![image](https://user-images.githubusercontent.com/65287080/154724563-8bb8d505-a7e6-4e34-878f-e1720a603adc.png)

**Comment by swhomans** [comment]
Excellent - I have been using v2.12 bootloader:
[bootloader4550V2-12.zip](https://github.com/PinguinoIDE/pinguinoide.github.io/files/8102210/bootloader4550V2-12.zip). The later version should work fine though.

**Reply by luftwaffe** [reply]
Thanks for the information, you can run Pinguino on windows 11

**Comment by callecuatro1976** [comment]
Cuando estará la versión actualizada para Windows????

**Reply by swhomans** [reply]
Lo siento, no lo se. No soy uno de los desarrolladores. La versión en la discusión anterior funciona bien si desea probarla. Disculpas por mi mal español!

**Comment by callecuatro1976** [comment]
No lo pude instalar, probé con el ide de la página y me llega hasta el error de pypip de ahí no puedo seguir!

**Comment by swhomans** [comment]
¿Ha utilizado mis instrucciones anteriores exactamente (18 de febrero)? Esto me funciona sin problema con windows 10/11. ¿Cuál es el error de Pypip exactamente?

**Comment by callecuatro1976** [comment]
Ya lo pude solucionar actualizando el pip

Enviado desde Correo<https://go.microsoft.com/fwlink/?LinkId=550986> para Windows

De: ***@***.***>
Enviado: miércoles, 13 de julio de 2022 13:30
Para: ***@***.***>
CC: ***@***.***>; ***@***.***>
Asunto: Re: [PinguinoIDE/pinguinoide.github.io] Support For windows and updated bootloader (Discussion #32)


¿Ha utilizado mis instrucciones anteriores exactamente (18 de febrero)? Esto me funciona sin problema con windows 10/11. ¿Cuál es el error de Pypip exactamente?

**Reply by callecuatro1976** [reply]
Gracias por las respuesta, lo solucione descargando la versión de la página de pinguino y actualice el pip por cmd y lo instalo sin problemas, ahora a usarlo a full, gracias

**Comment by swhomans** [comment]
excelente - dime si tienes mas problemas

**Reply by callecuatro1976** [reply]
Construi una placa y le grabe el booloader al 18f2550 y no me lo reconoce y no puedo subir los programas , alguna idea de que hacer?

**Reply by swhomans** [reply]
Creo que es un problema con el bootloader. ¿Tienes otra placa para comprobar que el código se está cargando correctamente desde Pinguino?

**Reply by callecuatro1976** [reply]
Sabes que era el capacitor vusb  le puse uno de 1uf y me lo reconoció, ahora se comunica con todo menos con el ide pinguino, ya voy a encontrar porque, creo que el bootloader para 18f2550  no es el correcto!

**Reply by swhomans** [reply]
sí, si se comunica con todo menos con el ide, creo que debe ser el bootloader. Lo siento, pero no he usado 2550 - no puedo decirte dónde encontrar el bootloader correcto.

**Comment by callecuatro1976** [comment]
pude hacerlo funcionar, me falta el libus archivo usb fue una odisea

**Reply by swhomans** [reply]
Muy bien. Creo que vale la pena :-)

**Comment by callecuatro1976** [comment]
Consulta, como cargo una librería en pinguino, quiero usar el dth11 y cargar librerías de arduino, se puede?

**Reply by swhomans** [reply]
Normalmente, yo copio las librerías nuevas en la carpeta correspondiente. Creo que no es posible utlizar librerías arduino directamente en pinguino.

**Comment by callecuatro1976** [comment]
Gracias se entendió muy bien

Obtener Outlook para Android<https://aka.ms/ghei36>
________________________________

**Comment by callecuatro1976** [comment]
consulta donde puedo encontrar algunos ejemplos de como usar pinguino bloques ?

**Comment by swhomans** [comment]
Hay una copia archivada del sitio web original de pinguino con ejemplos:

https://web.archive.org/web/20140328010344/http://wiki.pinguino.cc/index.php/Language_Reference


Contáctame si tienes algún problema.

Saludos,

Steve



Sent from Outlook<http://aka.ms/weboutlook>

________________________________

**Reply by callecuatro1976** [reply]
arme bastante con los bloques pinguino kit pero me cuelga el ide

**Comment by callecuatro1976** [comment]
)
![Captura de pantalla 2023-04-09 092442](https://user-images.githubusercontent.com/109005720/230772421-5c6b2fb7-2fd5-455f-9cda-f9a15591b106.png)

**Comment by swhomans** [comment]
¿Tu programa se compila en el IDE?



Sent from Outlook<http://aka.ms/weboutlook>

________________________________

**Reply by callecuatro1976** [reply]
Si compila y funciona muy bien, lo guardo y cuando quiero volver a abrirlo se desordena todos los bloques y no responde el ide

**Comment by swhomans** [comment]
Creo que es un problema con IDE Python. Lo siento - todavía yo utilizo v. X4rev959.




Sent from Outlook<http://aka.ms/weboutlook>

________________________________
